### ISSUE TITLE
Python int too large to convert to C unsigned long

### ISSUE THREAD

**Issue by gmatht** [borgbackup/borg#2319]
Using borg 1.1.0b3 and Ubuntu 16.04.2 I got the following error:

```
Local Exception2 GB C 143.90 GB D 919652 N 8/Users/John/AppData/Local/Temp/2b2cb.msi                                                                                                                                                                                                                                                                                                          
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/borg/archiver.py", line 2915, in main
    exit_code = archiver.run(args)
  File "/usr/local/lib/python3.5/dist-packages/borg/archiver.py", line 2843, in run
    return func(args)
  File "/usr/local/lib/python3.5/dist-packages/borg/archiver.py", line 106, in wrapper
    return method(self, args, repository=repository, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/borg/archiver.py", line 379, in do_create
    create_inner(archive, cache)
  File "/usr/local/lib/python3.5/dist-packages/borg/archiver.py", line 350, in create_inner
    read_special=args.read_special, dry_run=dry_run, st=st)
  File "/usr/local/lib/python3.5/dist-packages/borg/archiver.py", line 442, in _process
    read_special=read_special, dry_run=dry_run)
  File "/usr/local/lib/python3.5/dist-packages/borg/archiver.py", line 442, in _process
    read_special=read_special, dry_run=dry_run)
  File "/usr/local/lib/python3.5/dist-packages/borg/archiver.py", line 442, in _process
    read_special=read_special, dry_run=dry_run)
  File "/usr/local/lib/python3.5/dist-packages/borg/archiver.py", line 442, in _process
    read_special=read_special, dry_run=dry_run)
  File "/usr/local/lib/python3.5/dist-packages/borg/archiver.py", line 442, in _process
    read_special=read_special, dry_run=dry_run)
  File "/usr/local/lib/python3.5/dist-packages/borg/archiver.py", line 442, in _process
    read_special=read_special, dry_run=dry_run)
  File "/usr/local/lib/python3.5/dist-packages/borg/archiver.py", line 442, in _process
    read_special=read_special, dry_run=dry_run)
  File "/usr/local/lib/python3.5/dist-packages/borg/archiver.py", line 442, in _process
    read_special=read_special, dry_run=dry_run)
  File "/usr/local/lib/python3.5/dist-packages/borg/archiver.py", line 442, in _process
    read_special=read_special, dry_run=dry_run)
  File "/usr/local/lib/python3.5/dist-packages/borg/archiver.py", line 442, in _process
    read_special=read_special, dry_run=dry_run)
  File "/usr/local/lib/python3.5/dist-packages/borg/archiver.py", line 414, in _process
    status = archive.process_file(path, st, cache, self.ignore_inode)
  File "/usr/local/lib/python3.5/dist-packages/borg/archive.py", line 892, in process_file
    self.add_item(item)
  File "/usr/local/lib/python3.5/dist-packages/borg/archive.py", line 375, in add_item
    self.items_buffer.add(item)
  File "/usr/local/lib/python3.5/dist-packages/borg/archive.py", line 208, in add
    self.buffer.write(self.packer.pack(item.as_dict()))
  File "msgpack/_packer.pyx", line 223, in msgpack._packer.Packer.pack (msgpack/_packer.cpp:223)
  File "msgpack/_packer.pyx", line 225, in msgpack._packer.Packer.pack (msgpack/_packer.cpp:225)
  File "msgpack/_packer.pyx", line 195, in msgpack._packer.Packer._pack (msgpack/_packer.cpp:195)
  File "msgpack/_packer.pyx", line 140, in msgpack._packer.Packer._pack (msgpack/_packer.cpp:140)
OverflowError: Python int too large to convert to C unsigned long

Platform: Linux compaqt 4.4.0-59-generic #80-Ubuntu SMP Fri Jan 6 17:47:47 UTC 2017 x86_64 x86_64
Linux: Ubuntu 16.04 xenial
Borg: 1.1.0b3  Python: CPython 3.5.2
PID: 13917  CWD: /root/tmp3
sys.argv: ['/usr/local/bin/borg', 'create', '-s', '-p', '-C', 'lz4', '/media/gandalf/Seagate Expansion Drive/v11.borg::Lenovo1TB2016_AsFiles2', '.']
SSH_ORIGINAL_COMMAND: None

```

**Comment by gmatht** [comment]
BTW, I seem to have worked around this by using `-C zlib` instead.

**Comment by ThomasWaldmann** [comment]
@gmatht compression should be unrelated to this.

Some metadata of at least one of your files produces integers > 2^64. Often this is a broken timestamp.

I see you are using 1.1.0b3 and we recently fixed that in master branch, so could you try master?

https://github.com/borgbackup/borg/pull/2309

**Comment by ThomasWaldmann** [comment]
Maybe duplicate of #2304.

**Comment by ThomasWaldmann** [comment]
That Seagate disk, where your source files are on, is it NTFS?

**Comment by gmatht** [comment]
The seagate drive is the destination. It is NTFS, as are some of the source
partitions.

On 20 Mar 2017 12:16 PM, "TW" <notifications@github.com> wrote:

That Seagate disk, where your source files are on, is it NTFS?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/borgbackup/borg/issues/2319#issuecomment-287676192>,
or mute
the thread

**Comment by ThomasWaldmann** [comment]
Likely fixed by #2309 (also related: #2321).

**Comment by gmatht** [comment]
Just confirming that you were right, it failed eventually with -C zlib, but worked with the latest borg master (borg 1.1.0.dev9+ng16e6e3d).
